### PR TITLE
(PUP-11993) More Style cops!

### PR DESCRIPTION
**Pull request by AriaXLi**

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -647,6 +647,8 @@ Style/PreferredHashMethods:
   Enabled: false
 
 # This cop supports safe auto-correction (--auto-correct).
+# Enabling this cop will remove raising RuntimeErrors and cause spec test
+# failures
```
Comment (mhashizume) [inline .rubocop_todo.yml:651]: I think if the plan is to have this cop disabled indefinitely, we can probably move it from .rubocop_todo.yml to .rubocop.yml to indicate that it's not in some kind of pending state.

File: lib/puppet/provider/package/portage.rb [hunk 2]
```diff
@@ -121,7 +121,7 @@ def qatom
       else
         search = '=' + @resource[:name] + '-' + "#{should}"
       end
-      search_output = qatom_bin(*([search, '--format', output_format]))
```
Comment (mhashizume) [inline lib/puppet/provider/package/portage.rb:124]: Are these asterisks important to this provider?

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -672,6 +672,9 @@ Style/RedundantSelf:
   Enabled: false
 
 # This cop supports unsafe auto-correction (--auto-correct-all).
+# Enabling this cop causes a plethora of failed rspec tests, mostly
+# Errno::ENAMETOOLONG and Puppet::Context::UndefinedBindingError: Unable to
+# lookup 'environments' errors
```
Comment (mhashizume) [inline .rubocop_todo.yml:677]: Same comment here as Style/RedundantException

File: lib/puppet/provider/package/portage.rb (hunk 2 above)
Comment (AriaXLi) [reply lib/puppet/provider/package/portage.rb:124]: I believe the asterisk was removed as a part of the [Lint/RedundantSplatExpansion](https://msp-greg.github.io/rubocop/RuboCop/Cop/Lint/RedundantSplatExpansion.html) cop. With the extra parenthesis removed as a part of the Style/RedundantParenthesis cop, the line looks like:
`search_output = qatom_bin(*[search, '--format', output_format])`

Rubocop thinks the splat operator isn't needed in this case and each of those params can be passed in without it but tbh I'm not too familiar with the splat operator and I'm okay undoing that change if we're unsure.

**Review by mhashizume — APPROVED** [review]

**Pull request merged**